stan/math/prim/mat/fun/to_matrix.hpp

```diff
@@ -111,7 +100,162 @@ namespace stan {
             result(ij) = x[j][i];
         return result;
       } else {
-        return Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic> (0, 0);
+        return Eigen::Matrix<double, Eigen::Dynamic,
+                             Eigen::Dynamic> (0, 0);
+      }
+    }
+
+    /**
+     * Returns a matrix representation of the vector in column-major
+     * order with the specified number of rows and columns.
+     *
+     * @tparam T type of the scalar
+     * @param x matrix
+     * @param m rows
+     * @param n columns
+     * @return Reshaped inputted matrix
+     * @throw <code>std::invalid_argument</code> if the sizes
+     * do not match
+     */
+    template <typename T, int R, int C>
+    inline Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
+    to_matrix(const Eigen::Matrix<T, R, C>& x, int m, int n) {
+      static const char* fun = "to_matrix(matrix)";
+      check_size_match(fun, "rows * columns", m * n, "vector size",
+                       x.size());
+      Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> y = x;
+      y.resize(m, n);
+      return y;
+    }
+
+    /**
+     * Returns a matrix representation of the vector in column-major
+     * order with the specified number of rows and columns.
+     *
+     * @tparam T type of the scalar
+     * @param x vector of values
+     * @param m rows
+     * @param n columns
+     * @return the matrix representation of the input
+     * @throw <code>std::invalid_argument</code>
+     * if the sizes do not match
+     */
+    template <typename T>
+    inline Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>
+    to_matrix(const std::vector<T>& x, int m, int n) {
+      static const char* fun = "to_matrix(array)";
+      check_size_match(fun, "rows * columns", m * n, "vector size",
+                       x.size());
+      return Eigen::Map<const
+                        Eigen::Matrix<T, Eigen::Dynamic,
+                                      Eigen::Dynamic> >(&x[0], m, n);
+    }
+
+    /**
+     * Returns a matrix representation of the vector in column-major
+     * order with the specified number of rows and columns.
+     *
+     * @param x vector of values
+     * @param m rows
+     * @param n columns
+     * @return the matrix representation of the input
+     * @throw <code>std::invalid_argument</code>
+     * if the sizes do not match
+     */
+    inline Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>
+    to_matrix(const std::vector<int>& x, int m, int n) {
+      static const char* fun = "to_matrix(array)";
+      int size = x.size();
+      check_size_match(fun, "rows * columns", m * n,
+                       "vector size", size);
+      Eigen::Matrix<double,
+                    Eigen::Dynamic, Eigen::Dynamic> result(m, n);
+      double* datap = result.data();
```

Comment:
Indeed, it was probably slightly faster before check constraints of Eigen was disabled in Stan, but now it might just be the same thing, just making things harder to read.